Alloy/Encoders/RGBAToYCbCr.swift

```diff
@@ -0,0 +1,90 @@
+import Metal
+import MetalPerformanceShaders
+
+final public class RGBAToYCbCr {
+
+    // MARK: - Properties
+
+    public let pipelineState: MTLComputePipelineState
+    public let scale: MPSImageBilinearScale
+    private let deviceSupportsNonuniformThreadgroups: Bool
+
+    // MARK: - Life Cycle
+
+    public convenience init(context: MTLContext) throws {
+        try self.init(library: context.library(for: Self.self))
+    }
+
+    public init(library: MTLLibrary) throws {
+        self.deviceSupportsNonuniformThreadgroups = library.device
+                                                           .supports(feature: .nonUniformThreadgroups)
+        let constantValues = MTLFunctionConstantValues()
+        constantValues.set(self.deviceSupportsNonuniformThreadgroups,
+                           at: 0)
+        self.pipelineState = try library.computePipelineState(function: Self.functionName,
+                                                              constants: constantValues)
+        self.scale = .init(device: library.device)
+    }
+
+    // MARK: - Encode
+
+    public func callAsFunction(sourceRGBA: MTLTexture,
+                               destinationY: MTLTexture,
+                               destinationCbCr: MTLTexture,
+                               in commandBuffer: MTLCommandBuffer) {
+        self.encode(sourceRGBA: sourceRGBA,
+                    destinationY: destinationY,
+                    destinationCbCr: destinationCbCr,
+                    in: commandBuffer)
+    }
+
+    public func encode(sourceRGBA: MTLTexture,
+                       destinationY: MTLTexture,
+                       destinationCbCr: MTLTexture,
+                       in commandBuffer: MTLCommandBuffer) {
+        if destinationCbCr.size != destinationY.size {
```

Comment:
ooof, why do we want to make it so complicated? I think it is pretty straightforward that CbCr is always half of Y. Since we don't do any interleaving here, why would we want to support the case with equal size?

Reply:
AVFoundation in some cases needs equal sized planes